README.md

```diff
@@ -251,27 +251,70 @@ Every IPFS instance also exposes the libp2p API at `ipfs.libp2p`. The formal int
 
 ### [Runtime Support](https://github.com/ipfs/js-ipfs/issues/536)
 
+### Code Architecture and folder Structure
+
+![](/img/overview.png)
+
+##### Source code
+
+```Bash
+> tree src -L 2
+src                 # Main source code folder
+├── cli             # Implementation of the IPFS CLI
+│   └── ...
+├── http-api        # The HTTP-API implementation of IPFS as defined by http-api-spec
+├── core            # IPFS implementation, the core (what gets loaded in browser)
```

Comment:
My idea was to have `js-ipfs` (`ipfs` npm module) be a tiny wrapper that exposes the same functionality as it does now for end users. But code and tests would move into

- js-ipfs-core
- js-ipfs-cli
- js-ipfs-http

This way you get the same convenience but if you only want to use js-ipfs in the browser all you need is to install `js-ipfs-core`without all the cli + http baggage.